### PR TITLE
chore: Update k8s to v0.21.4

### PR DESCRIPTION
**Pull request by blkperl**
Signed-off-by: William Van Hevelingen <william.vanhevelingen@acquia.com>

Tips:

* Maybe add you organization to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md).
* Your PR needs to pass the required checks before it can be approved. If the check is not required (e.g. E2E tests) it does not need to pass
* Sign-off your commits to pass the DCO check: `git commit --signoff`.
* Run `make pre-commit -B` to fix codegen or lint problems. 
* Say how how you tested your changes. If you changed the UI, attach screenshots.
* If changes were requested, and you've made them, then dismis the review to get it looked at again.
* You can ask for help!

### CODE REVIEW

File: workflow/controller/operator_workflow_template_ref_test.go [hunk 1]
```diff
@@ -496,7 +496,7 @@ const wfTmplWithVol = `
 apiVersion: argoproj.io/v1alpha1
 kind: WorkflowTemplate
 metadata:
-  name: workflow-template-whalesay-template
+  name: workflow-template-whalesay-template-with-volume
```
Comment (blkperl) [inline workflow/controller/operator_workflow_template_ref_test.go:499]: I had to change the name to workaround this panic. It doesn't appear to change the behavior of the test though.

```
--- FAIL: TestWFTWithVol (0.00s)
panic: workflowtemplates.argoproj.io "workflow-template-whalesay-template" already exists [recovered]
	panic: workflowtemplates.argoproj.io "workflow-template-whalesay-template" already exists
```

**Review by alexec — COMMENTED** [review]
what's the benefit of this please?

**Comment by blkperl** [comment]
@alexec I don't think this there is a benefit beyond being one version higher. It seems to be backwards compatible with my local 1.20 cluster.

**Comment by alexec** [comment]
Lets hold this for v3.3

**Comment by wanghong230** [comment]
Could you do a validation in 0.21.4 for https://github.com/argoproj/argo-workflows/blob/master/examples/example-golang/main.go? 

We noticed that go-client 0.21.4 or go-client 0.22.x doesn't with argo 3.0, 3.1, 3.2. 
So the user to use the argo go client, has to use Kubernetes 0.20.x.

https://cloud-native.slack.com/archives/C01QW9QSSSK/p1630076431072800

**Comment by PrashantRaj18198** [comment]
When is this going to be merged?

**Comment by alexec** [comment]
We were planning to hold for v3.3, but maybe we need to bring it forward.  

This PR needs syncing with master.

**Comment by blkperl** [comment]
Synced with master

**Review by alexec — APPROVED** [review]

**Comment by alexec** [comment]
🚀

**Comment by sarabala1979** [comment]
https://github.com/argoproj/argo-workflows/issues/6750

**Pull request merged**